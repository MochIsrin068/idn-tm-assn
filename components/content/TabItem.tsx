import React from "react";

type TPropsTabItem = {
  label: string;
  onChangeTab: () => void;
  isActive: boolean;
};

export default function TabItem({
  label,
  onChangeTab,
  isActive,
}: TPropsTabItem) {
  return (
    <div
      className={`cursor-pointer flex items-center justify-center 
        py-2.5 
        px-8
        md:px-9
        rounded-3xl
        font-bold
      ${isActive ? "bg-red-200  text-red-500" : "border border-gray-400"}`}
      onClick={onChangeTab}
      data-testid="tab-button"
    >
      <span className="text-sm">{label}</span>
    </div>
  );
}
