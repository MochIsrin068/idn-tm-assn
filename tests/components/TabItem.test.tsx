import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import TabItem from "../../components/content/TabItem";

const mockProps = {
  label: "News",
  onChangeTab: jest.fn(),
  isActive: true,
};

describe("tests components TabItem", () => {
  it("Perform snapshots test TabItem", () => {
    const inputDefaultTree = render(<TabItem {...mockProps} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test text label", () => {
    render(<TabItem {...mockProps} />);
    const label = screen.getByText(/News/i);
    expect(label).toBeInTheDocument();
  });

  it("test text label with custom props", () => {
    render(<TabItem {...mockProps} label="Livestream" />);
    const label = screen.getByText(/Livestream/i);
    expect(label).toBeInTheDocument();
  });

  it("test button tab should call onChangeTab callback", () => {
    const onChangeTabSpy = jest.fn();

    const { getByTestId } = render(
      <TabItem {...mockProps} onChangeTab={onChangeTabSpy} />
    );

    fireEvent.click(getByTestId("tab-button"));

    expect(onChangeTabSpy).toHaveBeenCalled();
  });
});
