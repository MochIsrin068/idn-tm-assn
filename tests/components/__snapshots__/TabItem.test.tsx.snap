// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components TabItem Perform snapshots test TabItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cursor-pointer flex items-center justify-center 
        py-2.5 
        px-8
        md:px-9
        rounded-3xl
        font-bold
      bg-red-200  text-red-500"
        data-testid="tab-button"
      >
        <span
          class="text-sm"
        >
          News
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cursor-pointer flex items-center justify-center 
        py-2.5 
        px-8
        md:px-9
        rounded-3xl
        font-bold
      bg-red-200  text-red-500"
      data-testid="tab-button"
    >
      <span
        class="text-sm"
      >
        News
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
